### PR TITLE
feat: Add infraHostCleanShutdown to nrql alert conditions

### PR DESCRIPTION
**Pull request by Fallenstedt**
https://new-relic.atlassian.net/browse/NR-232402

NRQL Alert conditions will be receiving a new feature. Users will be able to use NRQL alert conditions to notify when New Relic stopped receiving data from an infrastructure agent.

The boolean on the expiration field will not trigger alerts for hosts that perform a clean shutdown option

This PR adds the boolean `infraHostCleanShutdown` to the expiration field. By default it is set to false at our API level. 

**Please do not merge yet**

### CODE REVIEW

File: pkg/alerts/nrql_conditions.go [hunk 1]
```diff
@@ -731,6 +732,7 @@ const (
       closeViolationsOnExpiration
       expirationDuration
       openViolationOnExpiration
+			infraHostCleanShutdown
     }
```
Comment (Fallenstedt) [inline pkg/alerts/nrql_conditions.go:736]: :memo: I have no idea why the indentation appears so off here. Locally, everything looks fine

**Comment by pranav-new-relic** [comment]
Hey @Fallenstedt - thanks for the contribution - we'll take a look at this soon :) currently juggling a couple of things

**Comment by Fallenstedt** [comment]
@pranav-new-relic 

I no longer work at New Relic. Please ping the dcon-hero in #help-alerts for updates on this PR.

**Comment by baNROne** [comment]
@pranav-new-relic We're reevaluating our approach to this project. Please close this PR for now.

**Pull request closed without merging**